src/ray/raylet/local_placement_group_manager_test.cc

```diff
@@ -0,0 +1,199 @@
+  init_unit_resource.insert({"CPU", 2.0});
+  ResourceSet init_resourece(init_unit_resource);
+  cluster_resource_map_[self_node_id_] = SchedulingResources(init_resourece);
+  local_available_resources_ = ResourceIdSet(init_resourece);
+  /// 3. prepare and commit two bundle resource.
+  local_placement_group_manager_->PrepareBundle(first_bundle_spec);
+  local_placement_group_manager_->PrepareBundle(second_bundle_spec);
+  local_placement_group_manager_->CommitBundle(first_bundle_spec);
+  local_placement_group_manager_->CommitBundle(second_bundle_spec);
+  /// 4. check remaining resources is correct after commit phase.
+  auto &remaining_resource = local_placement_group_manager_->GetAllResourceSetWithoutId();
+  std::vector<std::string> resource_labels = {"CPU_group_" + group_id.Hex(),
+                                              "CPU_group_1_" + group_id.Hex(),
+                                              "CPU_group_2_" + group_id.Hex()};
+  std::vector<double> resource_capacity = {2.0, 1.0, 1.0};
+  ResourceSet result_resource(resource_labels, resource_capacity);
+  ASSERT_EQ(3, local_available_resources_.AvailableResources().size());
+  ASSERT_EQ(1, remaining_resource.GetAvailableResources().IsEqual(result_resource))
+      << remaining_resource.GetAvailableResources().ToString() << " vs "
+      << result_resource.ToString();
+  ASSERT_EQ(1, local_available_resources_.ToResourceSet().IsEqual(result_resource))
+      << local_available_resources_.ToResourceSet().ToString() << " vs "
+      << result_resource.ToString();
+  /// 5. return second bundle.
+  local_placement_group_manager_->ReturnBundle(second_bundle_spec);
+  /// 6. check remaining resources is correct after return second bundle.
+  resource_labels = {"CPU", "CPU_group_" + group_id.Hex(),
+                     "CPU_group_1_" + group_id.Hex()};
+  resource_capacity = {1.0, 1.0, 1.0};
+  result_resource = ResourceSet(resource_labels, resource_capacity);
+  ASSERT_EQ(1, remaining_resource.GetAvailableResources().IsEqual(result_resource))
+      << remaining_resource.GetAvailableResources().ToString() << " vs "
+      << result_resource.ToString();
+  ASSERT_EQ(1, local_available_resources_.ToResourceSet().IsEqual(result_resource))
+      << local_available_resources_.ToResourceSet().ToString() << " vs "
+      << result_resource.ToString();
+  /// 7. return first bundel.
+  local_placement_group_manager_->ReturnBundle(first_bundle_spec);
+  /// 8. check remaining resources is correct after all bundle returned.
+  result_resource = ResourceSet(init_unit_resource);
+  ASSERT_EQ(1, remaining_resource.GetAvailableResources().IsEqual(result_resource))
+      << remaining_resource.GetAvailableResources().ToString() << " vs "
+      << result_resource.ToString();
+  ASSERT_EQ(1, local_available_resources_.ToResourceSet().IsEqual(result_resource))
+      << local_available_resources_.ToResourceSet().ToString() << " vs "
+      << result_resource.ToString();
+}
```

Comment:
Can you create an idempotency test here? what you can do is just to set the goal state and mix prepare/commit bundles in a random order multiple times and see if the goal state is correct. This is an important test for the protocol (since we retry prepare bundles or commit bundles until it succeeds). 

For example,

```
# Test if prepare bundles 10 times would work.
# Test if prepare -> commit -> prepare works.
# Test if prepare -> commit -> commit works
# Test if prepare -> return -> commit worrks
# Test if prepare -> return -> prepare works
```

You don't need to test all scenarios (although that's ideal).

Reply:
yeah, I have add all these case except the `# Test if prepare -> return -> prepare works` case , because it's just like `prepare bundle`